src/main/java/dev/joopie/jambot/listener/AutoCompleteBot.java

```diff
@@ -0,0 +1,43 @@
+package dev.joopie.jambot.listener;
+
+import dev.joopie.jambot.repository.artist.ArtistRepository;
+import dev.joopie.jambot.repository.track.TrackRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import net.dv8tion.jda.api.events.interaction.command.CommandAutoCompleteInteractionEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.dv8tion.jda.api.interactions.commands.Command;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class AutoCompleteBot extends ListenerAdapter {
+   private final ArtistRepository artistRepository;
+   private final TrackRepository trackRepository;
+
+   @Transactional
+   @Override
+    public void onCommandAutoCompleteInteraction(CommandAutoCompleteInteractionEvent event) {
```

Comment:
Er is een interface die je kunt implementeren op je command handler om autocomplete dingen te doen.

Reply:
Yes, but implementing this interface results in a lot of extra methods that I need to implement in that class. This is not ideal. At least, if we are talking about the same interface which is the `CommandAutoCompleteInteraction`

Reply:
Solved